### PR TITLE
[SOAR-9478] Extract it - Remove duplicated URLs from links

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Remove duplicated URLs from links

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by blaxminarayan-r7 — APPROVED** [review]

File: plugins/extractit/help.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ This action extracts all UUIDs, CVEs, dates, domains, emails, filepaths, IOCs, I
...
+|date_format|string|None|True|Dates matching this format are extracted - If All Formats is selected, found dates will be processed in the listed order documented|['dd/mm/yyyy', 'mm/dd/yyyy', 'dd\\mm\\yyyy', 'dd.mm.yyyy', 'dd-mm-yyyy', 'dd.mm.yy', 'dd-mm-yy', 'dd/mm/yy', 'dd\\mm\\yy', 'mm\\dd\\yyyy', 'mm.dd.yyyy', 'mm-dd-yyyy', 'mm/dd/yy', 'mm\\dd\\yy', 'mm.dd.yy', 'mm-dd-yy', 'dd/mmm/yyyy', 'dd\\mmm\\yyyy', 'dd.mmm.yyyy', 'dd-mmm-yyyy', 'dd/mmm/yy', 'dd\\mmm\\yy', 'dd.mmm.yy', 'dd-mmm-yy', 'yyyy.mm.dd', 'yyyy-mm-dd', 'yyyy/mm/dd', 'yyyy\\mm\\dd', 'yyyy.mmm.dd', 'yyyy-mmm-dd', 'yyyy/mmm/dd', 'yyyy\\mmm\\dd', 'yy.mm.dd', 'yy-mm-dd', 'yy/mm/dd', 'yy\\mm\\dd', 'yyyy-mm-ddThh:mm', 'yyyy-mm-ddThh:mm:ss', 'All Formats', '']|dd/mm/yyyy|
```
Comment (cmcnally-r7) [inline plugins/extractit/help.md:46]: Hey can you undo this change? It looks like when you regenerate and it adds extra characters.

if not then keep this :)

File: plugins/extractit/help.md [hunk 2]
```diff
@@ -189,7 +189,7 @@ This action extracts all dates from a string or file.
...
```
Comment (cmcnally-r7) [inline plugins/extractit/help.md:192]: Same here

File: plugins/extractit/icon_extractit/util/extractor.py [hunk 3]
```diff
@@ -149,6 +149,23 @@ def clear_domains(matches: list) -> list:
...
+    """
+    remove_extracted_urls_from_links This function allows to remove extracted links from URLs
+    :param input_list: List of extracted URLs
+    :return: List without additionally extracted URLs from link params
+    """
+    output_list = input_list.copy()
+    query_parameter_equals_character = "="
+    for url in input_list:
+        temporary_list = input_list.copy()
+        temp_url = f"{query_parameter_equals_character}{temporary_list.pop(temporary_list.index(url))}"
```
Comment (cmcnally-r7) [inline plugins/extractit/icon_extractit/util/extractor.py:162]: It's good code quality but its constants variable overkill 
```suggestion
        temp_url = f"={temporary_list.pop(temporary_list.index(url))}"
```
Equals sign is very common in URLs so I don't think we need a variable for it

File: plugins/extractit/icon_extractit/util/extractor.py [hunk 4]
```diff
@@ -149,6 +149,23 @@ def clear_domains(matches: list) -> list:
...
+        for url_check in temporary_list:
+            if temp_url.replace("&amp", "") in url_check:
+                output_list.pop(output_list.index(temp_url[len(query_parameter_equals_character) :]))
```
Comment (cmcnally-r7) [inline plugins/extractit/icon_extractit/util/extractor.py:165]: Can we have more context on this?
is there any case where it does **not** equal one?

**Review by cmcnally-r7 — COMMENTED** [review]
Just some small changes but looks good 👍

File: plugins/extractit/help.md (hunk 1 above)
Comment (igorski-r7) [reply plugins/extractit/help.md:46]: Unfortunately It has to stay like this - otherwise there will be validation error 😞
